### ISSUE TITLE
wrong validation happen for max-words validation class 

### ISSUE THREAD

**Issue by sunilit42** [magento/magento2#23538]
### Preconditions (*)
1. Latest Magento 2.3.2
2. create one form with validation class max-words

### Steps to reproduce (*)

1. Create one form with validation class max-words
e.g
```
<div class="control">
                 <input name="test" id="test"  class="input-text" type="text" data-validate="{required:true,'max-words':5}"/>
            </div>
```

### Expected result (*)

1.  if i set max-words 5 then it should not show error for 5 words
e.g
https://www.screencast.com/t/voeIZmbGM4

### Actual result (*)

1. it should not show error msg for 5 words

**Comment by sunilit42** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @sunilit42, here is your [Magento instance](https://i-23538-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23538-2-3-develop.instances.magento-community.engineering/admin](https://i-23538-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
@sunilit42 Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
@sunilit42 Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17923` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your report. 
The issue has been fixed in magento/magento2#23541 by @sunilit42 in 2.3-develop branch
Related commit(s):
 - [9e95177e73b029e8d3f1a40752d9f80d4df04405](https://github.com/magento/magento2/commit/9e95177e73b029e8d3f1a40752d9f80d4df04405)

The fix will be available with the upcoming 2.3.3 release.
